File: C#OOP/Polymorphism/Polymorphism-Lab/Shapes/Models/Circle.cs
using Shapes.Contracts;
using System;
using System.Collections.Generic;
using System.Text;

namespace Shapes.Models
{
    public class Circle : Shape, ICircle
    {
        public Circle(double radius)
        {
            Radius = radius;
        }
        public double Radius { get; private set;}

        public override double CalculateArea()
        {
            return Math.PI * Radius * Radius;
        }

        public override double CalculatePerimeter()
        {
            return 2 * Math.PI * Radius;
        }

        public override string Draw()
        {
            return $"{base.Draw()} {this.GetType().Name}";
        }
    }
}

File: JSAdvanced/ClassesАndAttributes-Lab/Circle.js
class Circle{
    constructor(r){
        this._radius = r;
    };
    
    get radius(){
        return this._radius;
    };

    set radius(number){
        this._radius = number;
    }
    get diameter(){
        return this._radius * 2;
    };
    
    set diameter(number){
        this._radius = number / 2;
    };

    get area(){
        return Math.PI * this._radius ** 2;
    };
}



let c = new Circle(2);
console.log(`Radius: ${c.radius}`);
console.log(`Diameter: ${c.diameter}`);
console.log(`Area: ${c.area}`);
c.diameter = 1.6;
console.log(`Radius: ${c.radius}`);
console.log(`Diameter: ${c.diameter}`);
console.log(`Area: ${c.area}`);
